### PR TITLE
Captions and transcripts delivery remarks corrections

### PR DESCRIPTION
**Pull request by sendr**

### CODE REVIEW

File: video_xblock/static/js/studio-edit.js [hunk 1]
```diff
@@ -226,7 +226,7 @@ function StudioEditableXBlock(runtime, element) {
             }
         })
     };
-    /** Replaces an existing transcript to transcriptsValue or adds new */
+    /** Replaces an existing transcript to transcriptsValue or adds new and return true if new one was added or false otherwise */
```
Comment (z4y4ts) [inline video_xblock/static/js/studio-edit.js]: Use [@returns notation](http://usejsdoc.org/tags-returns.html)

File: video_xblock/video_xblock.py [hunk 2]
```diff
@@ -323,8 +329,10 @@ def student_view(self, context=None):  # pylint: disable=unused-argument
         """
 
         player_url = self.runtime.handler_url(self, 'render_player')
-        transcript_download_link = self.get_transcript_download_link()
         download_transcript_handler_url = self.runtime.handler_url(self, 'download_transcript')
+        transcript_download_link = self.get_transcript_download_link()
+        full_transcript_download_link = \
+            (download_transcript_handler_url + transcript_download_link) if transcript_download_link else ''
```
Comment (z4y4ts) [inline video_xblock/video_xblock.py]: Use regular `if`. Inline on is hard to read here.

File: video_xblock/static/js/studio-edit.js [hunk 3]
```diff
@@ -226,8 +226,15 @@ function StudioEditableXBlock(runtime, element) {
...
-    var pushTranscript = function (lang, label, url, oldLang, $uploadButton){
+    /**
+     * Replaces an existing transcript to transcriptsValue or adds new
+     * Returns true if new one was added or false otherwise
+     * @param {String} lang
+     * @param {String} label
+     * @param {String} url
+     * @param {String} oldLang
+     */
+    var pushTranscript = function (lang, label, url, oldLang){
```
Comment (dorosh) [inline video_xblock/static/js/studio-edit.js]: var pushTranscript = function pushTranscript(lang, label, url, oldLang) 
add name to function

File: video_xblock/static/js/transcript-download.js [hunk 4]
```diff
@@ -0,0 +1,30 @@
+domReady(function(){
+  videojs('{{ video_player_id }}').ready(function(){
+    var player = this;
+    var transcripts = {{ player_state.transcripts_object }};
```
Comment (z4y4ts) [inline video_xblock/static/js/transcript-download.js]: `JSON.parse`?

File: video_xblock/static/js/videojs-transcript.js [hunk 5]
```diff
@@ -1,22 +1,22 @@
...
           enableTrack = true;
         } else {
           track.mode = 'disabled';
         }
-      };
-    };
-    if (!enableTrack && track.kind === 'captions') {
-      tracks[0].mode = 'showing';
-    }
+        if (!enableTrack && ind == arr.length - 1) {
```
Comment (z4y4ts) [inline video_xblock/static/js/videojs-transcript.js]: Use `===`

File: video_xblock/static/js/transcript-download.js (hunk 4 above)
Comment (sendr) [inline video_xblock/static/js/transcript-download.js]: done

File: video_xblock/static/js/videojs-transcript.js (hunk 5 above)
Comment (sendr) [inline video_xblock/static/js/videojs-transcript.js]: done

**Pull request merged**